README.md

```diff
@@ -17,63 +17,353 @@
+
+This example also shows the need for asynchronous actions. The action returns a Promise to indicate that the result of this action is not available yet when the function returns. Instead, the result is available in the request callback after the HTTP call completes, and is passed as an argument to the resolve() function.
+
+A call to `reject()` can be used to reject the Promise and signal that the activation has completed abnormally.
+
+2. Create an action from the `weather.js` file:
+
+```bash
+wsk action create weather weather.js
 ```
-To use as a docker action for Node.js 8
+
+3. Use the following command to run the action, and observe the output:
+
+```bash
+wsk action invoke --result weather --param location "Brooklyn, NY"
 ```
-wsk action update myAction myAction.js --docker openwhisk/action-nodejs-v8
+
+Using the `--result` flag means that the value returned from the action is shown as output on the command-line:
+
+```bash
+{
+    "msg": "It is 28 degrees in Brooklyn, NY and Cloudy!"
+}
 ```
-This works on any deployment of Apache OpenWhisk
 
-### To use on deployment that contains the rutime as a kind
-To use as a kind action using Node.js 6
+### Beyond Hello - JavaScript Promise for Queuing Asynchronous Actions
+
+You can add `then` together to transform values or run additional asynchronous actions one after another.
+
+The following example demonstrates multiple concpts:
+
+* Usage of `wsk.actions.invoke` to invoke any OpenWhisk action which returns `Promise`.
+* Adding `then` to replace `Hello` with `Goodbye` in the greeting message returned from action invoked inside of the `Promise`.
+
+1. Create `goodbye.js` with the following content:
+
+```javascript
+var openwhisk = require('openwhisk');
+
+function main(params) {
+    var wsk = openwhisk({ignore_certs: params.ignore_certs || false});
+    var wsk = openwhisk();
+    return wsk.actions.invoke({
+        actionName: '/whisk.system/samples/greeting',
+        params: {
+            name: params.name,
+            place: params.place
+        },
+        blocking: true
+    }).then(activation => {
+        console.log('activation:', activation);
+        var payload = activation.response.result.payload.toString();
+        var output = payload.replace("Hello", "Goodbye");
+        return { msg: output };
+    });
+}
 ```
-wsk action update myAction myAction.js --kind nodejs:6
+
+2. Create an action from the `goodbye.js` file:
+
+```bash
+wsk action create goodbye goodbye.js
 ```
-To use as a kind action using Node.js 8
+
+3. Use the following command to run the action, and observe the output:
+
+```bash
+wsk action invoke --result goodbye --param name "Amy" --param location "Paris"
 ```
-wsk action update myAction myAction.js --kind nodejs:8
+Returns:
+
+```bash
+{
+    "msg": "Goodbye, Amy from Paris!"
+}
 ```
 
-### Local development
-For Node.js 6
+### Node.js version 6 Enviornment
```

Comment:
Typo

Reply:
thanks @rabbah yup, I am evaluating how to separate these docs, this PR is first step towards adding new material. I am working on a separate PR to address splitting up docs.